composition-js/src/merging/merge.ts

```diff
@@ -418,6 +420,8 @@ class Merger {
     // Represent any applications of directives imported from these spec URLs
     // using @join__directive in the merged supergraph.
     this.joinDirectiveFeatureDefinitionsByIdentity.set(CONNECT_VERSIONS.identity, CONNECT_VERSIONS);
+    // Some federation directives are translated to @join__directive in the supergraph.
```

Comment:
nit: we don't translate it, `@join__directive` is for preserving target directive information in the supergraph